sys/dev/iscsi_initiator/iscsi.h

```diff
@@ -422,11 +422,21 @@ typedef struct {
      union ipdu_u	ipdu;
      u_int		hdr_dig;	// header digest
 
-     ahs_t		*ahs_addr;
+     union {
+#ifdef _KERNEL
+	     ahs_t * __capability ahs_uaddr;
+#endif
+	     ahs_t	*ahs_addr;
+     };
```

Comment:
I wonder if judicious use of macros could make these sorts of changes smaller without making them too confusing.

Reply:
I'm not sure.  In some cases I wonder if I I couldn't get by with making them __kerncap and just fixing some in-kernel uses to use __fromcap and __tocap casts, but usually it seemed like overloaded pointers were used too widely for that.